src/db.rs

```diff
@@ -125,13 +125,20 @@ fn migrate_v0_v1(txn: &Transaction) -> anyhow::Result<()> {
     // drop the old refs table, since the content can be extracted from blocks_v0
     txn.execute_batch("DROP TABLE IF EXISTS refs;")?;
     txn.execute_batch(INIT)?;
+    let mut stmt = txn.prepare("SELECT COUNT(*) FROM blocks_v0")?;
+    let num_blocks: i64 = stmt.query_row(NO_PARAMS, |r| r.get(0))?;
     let mut stmt = txn.prepare("SELECT * FROM blocks_v0")?;
     let block_iter = stmt.query_map(params![], |row| {
         Ok((row.get::<_, Vec<u8>>(2)?, row.get::<_, Vec<u8>>(3)?))
     })?;
     for (i, block) in block_iter.enumerate() {
         if i % 1000 == 0 {
-            info!("converting to new blocks, block {}", i);
+            info!(
+                "converting to new blocks, block {} of {} ({}%)",
+                i,
+                num_blocks,
+                i / (num_blocks as usize) * 100
```

Comment:
I have had a panic because of a division by zero when I did something similar. But in this case I guess it can not happen because if there are no blocks, it will not even get there.

Reply:
these are integers, though, so the multiplication needs to go first, right?

Reply:
both good points, fixed